src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -1247,6 +1251,28 @@ public void afterSave(@NonNull SCMNavigatorOwner owner) {
         try {
             // FIXME MINOR HACK ALERT
             StandardCredentials credentials = Connector.lookupScanCredentials((Item)owner, getApiUri(), credentialsId);
+            if (credentials instanceof GitHubAppCredentials) {
+                try {
+                    List<GHEvent> handledEvents = ((GitHubAppCredentials) credentials).getAppInstallation().getEvents();
+                    Method eventsM = GHEventsSubscriber.class.getMethod("events"); // TODO protected
+                    eventsM.setAccessible(true);
+                    boolean good = true;
+                    for (GHEventsSubscriber subscriber : GHEventsSubscriber.all()) {
+                        @SuppressWarnings("unchecked")
+                        Set<GHEvent> subscribedEvents = (Set<GHEvent>) eventsM.invoke(subscriber);
+                        if (!handledEvents.containsAll(subscribedEvents)) {
```

Comment:
This would make sure no events go unhandled :+1

Reply:
It might be too strict—to activate this patch you need to include the repository events which you might otherwise have skipped if you are only dealing with a small list of repositories—but I erred on the conservative side.